operator/pkg/controllers/controller.go

```diff
@@ -113,19 +115,6 @@ func (c *GenericController) removeFinalizer(ctx context.Context, resource Object
 		remainingFinalizers = append(remainingFinalizers, finalizer)
 	}
 	if len(remainingFinalizers) < len(resource.GetFinalizers()) {
-		if err := c.patchFinalizersToResource(ctx, resource, remainingFinalizers); err != nil {
-			return err
-		}
-		zap.S().Infof("Successfully deleted finalizer %s for cluster name %s", finalizerStr, resource.GetName())
-	}
-	return nil
-}
-
-func (c *GenericController) patchFinalizersToResource(ctx context.Context, resource Object, finalizers []string) error {
-	persisted := resource.DeepCopyObject()
-	resource.SetFinalizers(finalizers)
-	if err := c.Patch(ctx, resource, client.MergeFrom(persisted)); err != nil {
-		return fmt.Errorf("merging changes to kube object, %w", err)
+		resource.SetFinalizers(remainingFinalizers)
```

Comment:
consider using a set object https://github.com/deckarep/golang-set

Reply:
Went with this package https://pkg.go.dev/k8s.io/apimachinery/pkg/util/sets